backend/internal/role/handler.go

```diff
@@ -0,0 +1,407 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package role
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	serverconst "github.com/asgardeo/thunder/internal/system/constants"
+	"github.com/asgardeo/thunder/internal/system/error/apierror"
+	"github.com/asgardeo/thunder/internal/system/error/serviceerror"
+	"github.com/asgardeo/thunder/internal/system/log"
+	sysutils "github.com/asgardeo/thunder/internal/system/utils"
+)
+
+const handlerLoggerComponentName = "RoleHandler"
+
+// roleHandler is the handler for role management operations.
+type roleHandler struct {
+	roleService RoleServiceInterface
+}
+
+// newRoleHandler creates a new instance of roleHandler
+func newRoleHandler(roleService RoleServiceInterface) *roleHandler {
+	return &roleHandler{
+		roleService: roleService,
+	}
+}
+
+// HandleRoleListRequest handles the list roles request.
+func (rh *roleHandler) HandleRoleListRequest(w http.ResponseWriter, r *http.Request) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, handlerLoggerComponentName))
+
+	limit, offset, svcErr := parsePaginationParams(r.URL.Query())
+	if svcErr != nil {
+		handleError(w, logger, svcErr)
+		return
+	}
+
+	roleListResponse, svcErr := rh.roleService.GetRoleList(limit, offset)
+	if svcErr != nil {
+		handleError(w, logger, svcErr)
+		return
+	}
+
+	w.Header().Set(serverconst.ContentTypeHeaderName, serverconst.ContentTypeJSON)
+	w.WriteHeader(http.StatusOK)
+
+	isErr := writeToResponse(w, roleListResponse, logger)
+	if isErr {
+		return
+	}
+
+	logger.Debug("Successfully listed roles with pagination",
+		log.Int("limit", limit), log.Int("offset", offset),
+		log.Int("totalResults", roleListResponse.TotalResults),
+		log.Int("count", roleListResponse.Count))
+}
+
+// HandleRolePostRequest handles the create role request.
+func (rh *roleHandler) HandleRolePostRequest(w http.ResponseWriter, r *http.Request) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, handlerLoggerComponentName))
+
+	createRequest, err := sysutils.DecodeJSONBody[CreateRoleRequest](r)
+	if err != nil {
+		handleError(w, logger, &ErrorInvalidRequestFormat)
+		return
+	}
+
+	sanitizedRequest := rh.sanitizeCreateRoleRequest(createRequest)
+	createdRole, svcErr := rh.roleService.CreateRole(sanitizedRequest)
+	if svcErr != nil {
+		handleError(w, logger, svcErr)
+		return
+	}
+
+	w.Header().Set(serverconst.ContentTypeHeaderName, serverconst.ContentTypeJSON)
+	w.WriteHeader(http.StatusCreated)
+
+	isErr := writeToResponse(w, createdRole, logger)
+	if isErr {
+		return
+	}
+
+	logger.Debug("Successfully created role", log.String("role id", createdRole.ID))
```

Comment:
```suggestion
	logger.Debug("Successfully created role", log.String("roleId", createdRole.ID))
```

Shall we use camel case for log parameters for consistency?
Check other places as well